packages/xrpl/snippets/src/bridgeTransfer.ts

```diff
@@ -0,0 +1,166 @@
+/* eslint-disable @typescript-eslint/consistent-type-assertions -- needed here */
+/* eslint-disable no-await-in-loop -- needed here */
+import {
+  AccountObjectsRequest,
+  LedgerEntry,
+  Client,
+  XChainAccountCreateCommit,
+  XChainBridge,
+  XChainCommit,
+  XChainCreateClaimID,
+  xrpToDrops,
+  Wallet,
+  TransactionMetadata,
+} from '../../src'
+import { CreatedNode } from '../../src/models/transactions/metadata'
+
+async function sleep(sec: number): Promise<void> {
+  return new Promise((resolve) => {
+    setTimeout(resolve, sec * 1000)
+  })
+}
+
+const lockingClient = new Client(
+  'wss://sidechain-net1.devnet.rippletest.net:51233',
+)
+const issuingClient = new Client(
+  'wss://sidechain-net2.devnet.rippletest.net:51233',
+)
+const MAX_LEDGERS_WAITED = 5
+const LEDGER_CLOSE_TIME = 4
+
+void bridgeTransfer()
+
+async function bridgeTransfer(): Promise<void> {
+  await lockingClient.connect()
+  await issuingClient.connect()
+  const lockingChainDoor = 'rMAXACCrp3Y8PpswXcg3bKggHX76V3F8M4'
+
+  const accountObjectsRequest: AccountObjectsRequest = {
+    command: 'account_objects',
+    account: lockingChainDoor,
+    type: 'bridge',
+  }
+  const bridgeData = (await lockingClient.request(accountObjectsRequest)).result
+    .account_objects[0] as LedgerEntry.Bridge
+  const bridge: XChainBridge = bridgeData.XChainBridge
+  console.log(bridge)
+
+  const { wallet: wallet1 } = await lockingClient.fundWallet()
+  const wallet2 = Wallet.generate()
+
+  console.log(
+    `Creating ${wallet2.classicAddress} on the issuing chain via the bridge...`,
+  )
+
+  const fundTx: XChainAccountCreateCommit = {
+    TransactionType: 'XChainAccountCreateCommit',
+    Account: wallet1.classicAddress,
+    XChainBridge: bridge,
+    SignatureReward: bridgeData.SignatureReward,
+    Destination: wallet2.classicAddress,
+    Amount: (
+      parseInt(bridgeData.MinAccountCreateAmount as string, 10) * 2
+    ).toString(),
+  }
+  const fundResponse = await lockingClient.submitAndWait(fundTx, {
+    wallet: wallet1,
+  })
+  console.log(fundResponse)
+
+  // wait for the attestation to go through
+  let ledgersWaited = 0
+  let initialBalance = '0'
+  while (ledgersWaited < MAX_LEDGERS_WAITED) {
+    await sleep(LEDGER_CLOSE_TIME)
+    try {
+      initialBalance = await issuingClient.getXrpBalance(wallet2.classicAddress)
+      console.log(
+        `Wallet ${wallet2.classicAddress} has been funded with a balance of ${initialBalance} XRP`,
+      )
+      break
+    } catch (_error) {
+      ledgersWaited += 1
+      // eslint-disable-next-line max-depth -- needed here
+      if (ledgersWaited === MAX_LEDGERS_WAITED) {
+        throw Error('Destination account creation via the bridge failed.')
```

Comment:
Is there a way to get a more meaningful error on why the account creation may have failed? (Ex. Querying the witness server to ask what happened to your transaction?)

Reply:
There isn't any way to query the witness servers directly - for security reasons, they aren't public. The best I could do is figure out the signing key that didn't submit an attestation, but that seems like rather detailed information to look for in a snipppet.

Reply:
I was more wondering since I can see this waiting + error logic being a helper function in the library, and having more robust error messaging for this kind of transaction seems helpful for end users if we can invest in building out that debugging logic ahead of time.

Reply:
It seems a bit complex to put that logic into the library itself, which is why I just wrote a snippet for it. Devs can copy-paste it into their own code.